include/seqan3/alignment/matrix/detail/alignment_score_matrix_one_column.hpp

```diff
@@ -0,0 +1,192 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::alignment_score_matrix_one_column.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/alignment/matrix/detail/alignment_matrix_column_major_range.hpp>
+#include <seqan3/alignment/matrix/detail/alignment_score_matrix_one_column_base.hpp>
+#include <seqan3/alignment/matrix/detail/alignment_score_matrix_proxy_type.hpp>
+#include <seqan3/core/concept/core_language.hpp>
+#include <seqan3/core/simd/concept.hpp>
+#include <seqan3/std/iterator>
+#include <seqan3/std/ranges>
+#include <seqan3/std/span>
+
+namespace seqan3::detail
+{
+
+/*!\brief An alignment score matrix storing only a single column for the computation.
```

Comment:
Can it be called matrix then? 

And logically: how can a score matrix be reduced to less than _alphabet_size * (alphabet_size + 1) / 2_ entries, if you have to store the scores for all combinations of characters??

Reply:
okay 👍